java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -945,6 +945,12 @@ class DatabaseImpl(
     db: SQLiteDatabase
   ): Unit = stats.delete.timeSuspending { counters ->
     db.transaction {
+      if (storageKey is InlineStorageKey) {
+        throw UnsupportedOperationException(
+          "Invalid attempt to delete inline storage key ${storageKey.toString()}." +
+          " Inline entities should not be removed using delete()."
```

Comment:
why not?
insertInlineEntity() calls createEntityStorageKeyId with an InlineStorageKey. then createEntityStorageKeyId calls delete. so I think this method is being called with inline keys. Why is that a problem?

Reply:
hmm, I might need to add a flag then. It's a problem to call this on an inline key if you're not either replacing the entity or deleting the parent somehow, because it leaves the parent in an inconsistent state.